Architecture_Engine/Modify/SetGeometry.cs

```diff
@@ -42,17 +45,67 @@ public static partial class Modify
         [Output("room", "An Architecture Room with an updated geometry")]
         public static Room SetGeometry(this Room room, Point locationPoint = null, ICurve perimeterCurve = null)
         {
-            if(room == null)
+            if (room == null)
             {
                 BH.Engine.Reflection.Compute.RecordError("Cannot set the geometry of a null room.");
                 return room;
             }
 
             Room clone = room.DeepClone();
             clone.Location = locationPoint == null ? room.Location.DeepClone() : locationPoint;
-            clone.Perimeter = perimeterCurve == null ? room.Perimeter.DeepClone() : perimeterCurve; 
+            clone.Perimeter = perimeterCurve == null ? room.Perimeter.DeepClone() : perimeterCurve;
             return clone;
         }
+
+        /***************************************************/
+
+        [Description("Assign new geometry (location) to a BuildersWork Opening. Dimensions need to be set by updating the properties of Opening's profile.")]
+        [Input("opening", "A BuildersWork Opening to set the geometry (location) of.")]
+        [Input("locationPoint", "A BHoM geometry Point defining the new location of the Opening.")]
+        [Output("opening", "A BuildersWork Opening with updated geometry (location).")]
+        public static Opening SetGeometry(this Opening opening, Point locationPoint)
+        {
+            if (opening == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Cannot set the geometry of a null opening.");
+                return null;
+            }
+
+            if (locationPoint == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Cannot set the geometry of an opening to a null point.");
+                return null;
+            }
+
+            Opening clone = opening.ShallowClone();
+            clone.CoordinateSystem = new Cartesian(locationPoint, opening.CoordinateSystem.X, opening.CoordinateSystem.Y, opening.CoordinateSystem.Z);
+
+            BH.Engine.Reflection.Compute.RecordWarning("Only the location point of the opening has been updated - to update its dimension, please modify relevant properties of the Profile.");
+            return clone;
+        }
+
+        /***************************************************/
+
+        [Description("Assign new geometry (location) to a BuildersWork Opening. Dimensions need to be set by updating the properties of Opening's profile.")]
+        [Input("opening", "A BuildersWork Opening to set the geometry (location) of.")]
+        [Input("coordinateSystem", "A BHoM geometry Cartesian coordinate system defining the new location (incl. orientation) of the Opening.")]
+        [Output("opening", "A BuildersWork Opening with updated geometry (location).")]
+        public static Opening SetGeometry(this Opening opening, Cartesian coordinateSystem)
```

Comment:
Is a coordinate system geometry? I know it sits in the geometry oM but is it generally considered geometry in the same way as a point or line?

Reply:
Dunno, sounds like a philosophic question to me. I added the `Cartesian` overload because the opening's location in space is represented by it, I simply thought it is no harm and better have than not have. I hope this makes sense 😛